pytube/__main__.py

```diff
@@ -102,6 +103,34 @@ def __init__(
             self.prefetch()
             self.descramble()
 
+    def check_availability(self):
+        """Check whether the video is available.
+        Raises different exceptions based on why the video is unavailable,
+        otherwise does nothing.
+
+        """
+        if self.watch_html is None:
+            raise VideoUnavailable(video_id=self.video_id)
+
+        status, messages = extract.playability_status(self.watch_html)
+        for reason in messages:
+            if status == 'UNPLAYABLE':
+                if reason == (
+                    'Join this channel to get access to members-only content '
+                    'like this video, and other exclusive perks.'
+                ):
```

Comment:
Warning! These reason checks will not work for everybody! These messages are generated based on the location of the IP address of the requesting client. If your IP address has been geo-located by Google/Youtube to be in an english speaking country, it will indeed yield these messages, however, it will change the language if the IP originates from other countries!

Reply:
https://github.com/pytube/pytube/blob/4faaa5fe8d88abd7b6fbaaab5dc287f0a3f5c502/pytube/request.py#L19
We have this accounted for in the way that we fetch videos -- we explicitly tell YouTube to give us english-language results for this.

Reply:
Awesome! I did not notice that. Nevermind then. Thanks! :)